shared/Microsoft.Extensions.StackTrace.Sources/StackFrame/StackTraceHelper.cs

```diff
@@ -119,5 +144,101 @@ internal static MethodDisplayInfo GetMethodDisplayString(MethodBase method)
 
             return methodDisplayInfo;
         }
+
+        private static bool ShowInStackTrace(MethodBase method)
+        {
+            Debug.Assert(method != null);
+
+            // Don't show any methods marked with the StackTraceHiddenAttribute
+            // https://github.com/dotnet/coreclr/pull/14652
+            foreach (var attibute in method.CustomAttributes)
+            {
+                // internal Attribute, match on name
+                if (attibute.AttributeType.Name == "StackTraceHiddenAttribute")
+                {
+                    return false;
+                }
+            }
+
+            var type = method.DeclaringType;
+            if (type == null)
+            {
+                return true;
+            }
+
+            foreach (var attibute in type.CustomAttributes)
+            {
+                // internal Attribute, match on name
+                if (attibute.AttributeType.Name == "StackTraceHiddenAttribute")
+                {
+                    return false;
+                }
+            }
+
+            // Fallbacks for runtime pre-StackTraceHiddenAttribute
+            if (type == typeof(ExceptionDispatchInfo) && method.Name == "Throw")
+            {
+                return false;
+            }
+            else if (type == typeof(TaskAwaiter) ||
+                     type == typeof(TaskAwaiter<>) ||
```

Comment:
We don't try to align things like this in our code (harder to maintain). Could you let the editor naturally tab these?

Reply:
Fixed